### PR TITLE
Integrate remote-settings-client

### LINKED ISSUE #221: Integrate remote-settings-client in Merino (in a branch)
**Acceptance criteria**

* Merino should compile 
* The remote-settings-client should be able to connect to the RS servers (even without downloading attachments)
* All the non RS related tests should pass.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/CONSVC-1374)

### PR DESCRIPTION
**Pull request by Dexterp37**
This fixes #221 and contributes to #218 .

This does not currently enable signature verification, but exclusively replaces the custom rs implementation with the remote-settings-client one.

### CODE REVIEW

File: merino-adm/src/remote_settings/mod.rs [hunk 1]
```diff
@@ -43,19 +42,28 @@ impl RemoteSettingsSuggester {
...
-                .as_ref()
-                .unwrap_or(&settings.remote_settings.default_bucket)
-                .clone(),
-            config
-                .collection
-                .as_ref()
-                .unwrap_or(&settings.remote_settings.default_collection)
-                .clone(),
-        )?;
+        let mut remote_settings_client = remote_settings_client::Client::builder()
```
Comment (pjenvey) [inline merino-adm/src/remote_settings/mod.rs:45]: Previously we were utilizing a `FileStorage`, e.g. `remote_settings_client.storage(Box::new(FileStorage..`, do we want to do that again? cc @mythmon

Comment (mythmon) [reply merino-adm/src/remote_settings/mod.rs:45]: Yes, if we don't we'll be re-downloading attachments at every sync.

File: merino-adm/src/remote_settings/reqwest_client.rs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+impl RsRequester for ReqwestClient {
+    async fn get(&self, url: RsUrl) -> Result<RsResponse, ()> {
+        match self
+            .reqwest_client
+            .get(url.clone())
+            .timeout(std::time::Duration::from_secs(3))
+            .send()
+            .await
+            .and_then(Response::error_for_status)
+            .context(format!("Fetching records from remote settings: {}", url))
```
Comment (mythmon) [inline merino-adm/src/remote_settings/reqwest_client.rs:40]: This is used for more than just fetching records. We could change this log line to "Performing HTTP request for Remote Settings", or similar.

File: merino-adm/src/remote_settings/reqwest_client.rs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                let body = match response.bytes().await {
+                    Err(e) => {
+                        tracing::error!(
+                            "ReqwestClient - unable to parse response body. {:?}",
+                            e.to_string()
+                        );
+                        return Err(());
+                    }
+                    Ok(b) => b,
+                };
```
Comment (mythmon) [inline merino-adm/src/remote_settings/reqwest_client.rs:67]: Would something like 

```rust
let body = response.bytes().await
	.map_err(|err| {
		tracing::error!(...);
		()
	})?;
```

Work? Using `map_err` to both log the messages and transform the error into `()` helps re-enable the Try operator, while keeping the same behavior.

File: merino-integration-tests/src/utils/test_tools.rs [hunk 4]
```diff
@@ -235,7 +235,13 @@ impl TestReqwestClient {
 async fn setup_remote_settings_bucket(server: &str) -> Result<(String, String)> {
     let reqwest_client = reqwest::Client::new();
     let bucket_info: serde_json::Value = reqwest_client
-        .post(format!("{}/v1/buckets/", server))
+        // Unfortunately `/dev/kinto.ini` does not support `/buckets/*`
+        // in `kinto.changes.resources`, so we should use the same bucket
+        // (i.e. `/buckets/main`) to create our collections in and make sure
+        // remote settings behave as expected. We use PUT here to overwrite
+        // the bucket transparently each time, since we don't really care
+        // about the stored metadata.
```
Comment (mythmon) [inline merino-integration-tests/src/utils/test_tools.rs:243]: Will calling `PUT` here delete the underlying collections? This method gets called as a part of parallel test runs, so we don't want to delete the collections used by other tests. The integration tests are one of the primary users of `/dev/kinto.ini`, so we can change that to support the old method, if that's possible.

**Review by mythmon — CHANGES_REQUESTED** [review]

File: merino-integration-tests/src/utils/test_tools.rs (hunk 4 above)
Comment (Dexterp37) [reply merino-integration-tests/src/utils/test_tools.rs]: Good point. I restored the POST

**Review by mythmon — APPROVED** [review]

File: merino-integration-tests/src/utils/test_tools.rs (hunk 4 above)
Comment (leplatrem) [reply merino-integration-tests/src/utils/test_tools.rs]: > Will calling PUT here delete the underlying collections?

No it won't

**Pull request merged**